cmd/cache.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright © 2020 Hirohito Sasakawa
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	bq "cloud.google.com/go/bigquery"
+
+	"github.com/hirosassa/bqiam/metadata"
+	"github.com/spf13/cobra"
+	"google.golang.org/api/bigquery/v2"
+	"google.golang.org/api/iterator"
+)
+
+// cacheCmd represents the cache command
+var cacheCmd = &cobra.Command{
+	Use:   "cache",
+	Short: "cache fetches bigquery datasets' metadata and store it in local file",
+	Long:  `cache fetches bigquery datasets' metadata and store it in local file`,
+	Run: func(cmd *cobra.Command, args []string) {
+		runCmdCache()
+	},
+}
+
+func runCmdCache() {
+	var metas metadata.Metas
+	projects, _ := listProjects()
+	for _, p := range *projects {
+		ds, _ := listDataSets(p)
+		for _, d := range *ds {
+			projectMetas, _ := listMetaData(p, d)
+			metas.Metas = append(metas.Metas, projectMetas.Metas...)
+		}
+	}
+	err := metas.Save(config.CacheFile)
+	if err != nil {
+		fmt.Println("Failed to fetch cache: %s", err)
```

Comment:
You mean `Faile to save cache` ?

and, need to return ?

Is this message below correct?
```
$ ./bqiam cache

Failed to fetch cache: %s Failed to save meta data cache file. err: open ~/.bqiam-cache-file.toml: no such file or directory
dataset meta data are cached to %s ~/.bqiam-cache-file.toml
```

Reply:
You're right! I fixed this implementation. Please check

Reply:
@hirosassa LGTM. Thanks!